helpers/DeployHelper.ts

```diff
@@ -61,16 +61,24 @@ export class DeployHelper {
     const stateLib = await this.deployStateLib();
 
     this.log("deploying stateV2...");
-    const StateV2Factory = await ethers.getContractFactory("StateV2", {
+    const StateV2Factory = await ethers.getContractFactory(stateContractName, {
       libraries: {
         StateLib: stateLib.address,
         SmtLib: smtLib.address,
         PoseidonUnit1L: poseidon1Elements.address,
       },
     });
-    const stateV2 = await upgrades.deployProxy(StateV2Factory, [verifier.address, NetworkIdTypes.polygonMumbai], {
-      unsafeAllowLinkedLibraries: true,
-    });
+    let stateV2;
+    if (isDeployedStateContract) {
+      stateV2 = await upgrades.deployProxy(StateV2Factory, [verifier.address], {
+        unsafeAllowLinkedLibraries: true,
+      });
+    } else {
+      stateV2 = await upgrades.deployProxy(StateV2Factory, [verifier.address, NetworkIdTypes.polygonMumbai], {
```

Comment:
Why don't we use hardcoded NetworkIdTypes.polygonMumbai?
What if I would deploy it to other network? Will it go with other id type?